### PR TITLE
add: init functionality

### PR DESCRIPTION
**Pull request by Inkvisto**
Hi everyone who clicked on link :)

This project presents the Rust Tracking Program, a logging tool crafted to assist users in effectively tracking tasks and their corresponding elapsed time. By offering a systematic approach to logging activities, the program seeks to improve productivity monitoring.

🛠️ Purpose

The Rust Tracking Program was developed to offer users a streamlined method for tracking tasks, including logging start times and durations. It addresses the need for an organized approach to managing daily activities and assessing productivity over time.

⚙️ How It Works

The Rust Tracking Program allows users to log tasks easily, with each entry containing a timestamp, a description of the task, and the duration it took to complete. Users can append new entries to an existing log file, delete specific entries, and dynamically adjust output to fit their terminal width. The program is designed to provide a user-friendly experience while ensuring data persistence between sessions.

Itracker

💻 Technology Stack

    🦀 Rust – as the primary programming language used for the project.
    📅 Chrono – a date and time library for managing timestamps and durations.
    📜 Clap – a command-line argument parser for handling command-line options and flags.
    🗂️ Config – a library for managing configuration files.
    🖥️ Crossterm – a cross-platform library for terminal manipulation, facilitating user input and output.
    🎨 Ratatui – a library for building rich text user interfaces, providing layout and rendering tools.
    🔄 Serde – a framework for serializing and deserializing Rust data structures.
    📄 TOML – a parser for reading configuration files.
    ✏️ Tui-textarea – a library for creating text areas in terminal user interfaces, enhancing user input capabilities.

The first version of this project was developed as a capstone project at the Ukrainian Rust Community Bootcamp. Your feedback is invaluable in refining this tool further.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /target
+Cargo.lock
```
Comment (Citizen4our) [inline .gitignore]: I assume you forget remove cargo.lock from git.

**Review by FairlyBet — APPROVED** [review]
The code looks clean. I like it.

File: src/log.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+/// Reads logs from a specified file and returns a vector of `LogEntry`.
+///
+/// # Arguments
+/// - `file_path`: The path to the log file.
+///
+/// # Returns
+/// - `Ok(Vec<LogEntry>)`: A vector of log entries if successful.
+/// - `Err(std::io::Error)`: An error if file operations fail.
+pub fn read_logs_from_file(file_path: &str) -> Result<Vec<LogEntry>, std::io::Error> {
```
Comment (Citizen4our) [inline src/log.rs:23]: Maybe better use some other format for log activity, like CSV, jsonx etc.? It is simplify in general logic for read/write/delete in this file, I think

**Review by Citizen4our — APPROVED** [review]
Code looks good and documented)

File: Readme.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+## Technologies Used
+1. **Rust**: The primary programming language used for the project.
+2. **Chrono**: A date and time library for Rust, used for managing timestamps and durations.
+3. **Clap**: A command-line argument parser for Rust, allowing for easy handling of command-line options and flags.
+4. **Config**: A configuration library for Rust, enabling the management of configuration files.
+5. **Crossterm**: A cross-platform library for terminal manipulation, used for handling user input and terminal output.
+6. **Ratatui**: A library for building rich text user interfaces in Rust, providing tools for layout and rendering.
+7. **Serde**: A framework for serializing and deserializing Rust data structures, enabling the conversion between Rust types and formats like JSON and TOML.
+8. **TOML**: A parser for TOML files, used for reading configuration files.
+9. **Tui-textarea**: A library for creating text areas in terminal user interfaces, facilitating user input.
```
Comment (Citizen4our) [inline Readme.md:26]: Add please some screenshots in README or PR description)

File: src/main.rs [hunk 4]
```diff
@@ -1,28 +1,200 @@
...
-    let args = Args::parse();
-    println!("{args:?}");
+    // Parse command line arguments
+    let args = parse_args();
+
+    // Handle log deletion if specified
+    if let Some(index) = args.delete_log {
+        let log_file = args.log.as_deref().unwrap_or("logs.txt");
+        log::delete_log_entry(log_file, index)?;
+        println!("Log entry at index {} deleted from {}.", index, log_file);
```
Comment (BigglesworthCat) [inline src/main.rs:28]: You can write this like:
```
        println!("Log entry at index {index} deleted from {log_file}.");
```

File: src/main.rs [hunk 5]
```diff
@@ -1,28 +1,200 @@
...
+fn read_start_time_and_paused_duration_from_csv(
+    output_file: &str,
+    index: usize,
+) -> Result<(SystemTime, Duration), std::io::Error> {
+    let file = OpenOptions::new().read(true).open(output_file)?;
+    let mut reader = ReaderBuilder::new().from_reader(BufReader::new(file));
+
+    for result in reader.records() {
+        let record = result?;
+        if record.len() >= 5 {
```
Comment (BigglesworthCat) [inline src/main.rs:158]: Magic number

**Review by BigglesworthCat — APPROVED** [review]
I like the idea of your project. I like its project structure. I like its tech stack.
I think, it can be useful, if you'll develop this project.
In my opinion, fully TUI application will be more user friendly and will eliminate the need for repeated commands to change the state of the timer and tasks.

Good job👍

But there is some flaws.

It would be nice to have `README.md` with detailed information how to run this program. Maybe it is a good idea to represent synopsis of a program there.

Also, as I understood, there is some тще obvious behavior with default values of log and config files, default values of `Args` if it has errors (after invalid arguments).

Also, if would be great to use `clippy` to reduce redundant or non-conventional blocks/statements.

**Pull request merged**